HZETRN/btrn.py

```diff
@@ -0,0 +1,377 @@
+    - Get the reference point and determine the start volume
+    - For each ray direction, starting from the start vol
+      o get the list of materials and distances it passes through
+      o create the contents of the transport input file
+"""
+def main():
+    global spatial_dir
+    global config_log
+
+    # Setup: parse the the command line parameters
+    args = parsing()
+    uwuw_file = os.path.join(os.path.dirname('__file__'), args.uwuw_file)
+
+    cur_path = os.path.dirname(os.path.abspath(__file__)) + '/'
+    run_path = cur_path + args.run_dir + '/'
+    config_path = run_path + config_log
+
+    logging.basicConfig(filename=config_path, level=logging.INFO, format='%(asctime)s %(message)s')
+    message = 'python btrn.py -f ' + args.uwuw_file + \
+              ' -d ' + args.run_dir + \
+              ' -e ' + args.rad_env + \
+              ' -r ' + ('not given' if not args.ray_dir_file else args.ray_dir_file)
+    logging.info(message)
+
+    # Ensure needed template files exist
+    one_d_tool.check_temp_files(run_path)
+
+    # Transport results for each direction will be placed in this directory:
+    # Ensure it exists before proceeding.
+    data_path = run_path + 'data/'
+    if not os.path.isdir(data_path):
+	os.mkdir(data_path)
+
+    rad_env_file = data_path + 'rad_env.txt'
+    with open(rad_env_file, 'w') as f:
+        f.write(args.rad_env)
+
+    spatial_path = cur_path + spatial_dir + '/'
+    if not os.path.isdir(spatial_path):
+	os.mkdir(spatial_path)
+
+    # Load the DAG object for this geometry
+    rtn = dagmc.load(uwuw_file)
+
+    vol_fname_dict = tag_utils.get_fnames_for_vol(uwuw_file)
+    ray_tuples = get_directions(args.ray_dir_file, args.rand_dirs)
+
+    # The default starting point is 0,0,0
+    ref_point = load_ray_start(args.ray_start)
+    start_vol = find_ref_vol(ref_point)
+    
+    for dir in ray_tuples:
+	###################################
+	# For backscatter:  
+	# Create two lists 'a' and 'b'.  
+	#    'a' is the list of slab materials and lengths in the given direction.  
+	#    'b' is the list of same items going OPPOSITE (180 degrees)
+	print start_vol, ref_point, "dir", dir
+	slab_lens_a, slab_mat_names_a = slabs_for_ray(start_vol, ref_point,  dir, vol_fname_dict)
+	slab_lens_b, slab_mat_names_b = slabs_for_ray(start_vol, ref_point, -dir, vol_fname_dict)
+
+	# reverse order and add backward dir
+        # slab_lens = slab_lens_a[::-1] + slab_lens_b
+        slab_lens      = slab_lens_b[::-1]      + slab_lens_a
+	slab_mat_names = slab_mat_names_b[::-1] + slab_mat_names_a
+	num_slabs_to_ref = len(slab_lens_b)
+	num_mats = len(slab_mat_names)
+        #############################################
+	# Create the transport geometry file contents
+        # Need to have a non-zero number of mats
+	transport_input = []
+	if 0 != num_mats:
+	    effective_num_mats = num_mats
+	    last_mat_name = slab_mat_names[0]
+	    last_slab_points = [0.0, slab_lens[0]]
+
```

Comment:
create a function called create_input_file, maybe should live in hztran specific location. make_file text for example